xarray/tests/test_missing.py

```diff
@@ -559,6 +633,114 @@ def test_bfill_dataset(ds):
     ds.ffill(dim="time")
 
 
+def test_get_gap_left_edge():
+    n = np.nan
+    arr = [
+        [n, 1, n, n, n, n, n, n, 4],
+        [n, n, n, 1, n, n, 4, n, n],
+    ]
+
+    y = np.arange(9) * 3
+    da = xr.DataArray(arr, dims=["x", "y"], coords={"x": [0, 1], "y": y})
+    index = get_clean_interp_index(da, dim="y", use_coordinate=True)
+    actual = _get_gap_left_edge(da, dim="y", index=index)
+    expected = da.copy(
+        data=[[n, 3, 3, 3, 3, 3, 3, 3, 24], [n, n, n, 9, 9, 9, 18, 18, 18]]
+    )
+    assert_equal(actual, expected)
+
+    actual = _get_gap_left_edge(da, dim="y", index=index, outside=True)
+    expected = da.copy(
+        data=[[0, 3, 3, 3, 3, 3, 3, 3, 24], [0, 0, 0, 9, 9, 9, 18, 18, 18]]
+    )
+    assert_equal(actual, expected)
+
+    y = [0, 2, 5, 6, 7, 8, 10, 12, 14]
+    da = xr.DataArray(arr, dims=["x", "y"], coords={"x": [0, 1], "y": y})
+    index = get_clean_interp_index(da, dim="y", use_coordinate=True)
+    actual = _get_gap_left_edge(da, dim="y", index=index)
+    expected = da.copy(
+        data=[[n, 2, 2, 2, 2, 2, 2, 2, 14], [n, n, n, 6, 6, 6, 10, 10, 10]]
+    )
+
+
+def test_get_gap_right_edge():
+    n = np.nan
+    arr = [
+        [n, 1, n, n, n, n, n, n, 4],
+        [n, n, n, 1, n, n, 4, n, n],
+    ]
```

Comment:
(FWIW this is a really nice way of aligning arrays, haven't thought of this before)